### PR TITLE
Fixes for current Hearthstone patch level

### PR DESCRIPTION
**Pull request by djkaty**
Fix first .PegasusShared package is now DatabaseDeckCard in types
Fix incorrect directory path in auto-generated import statements when outputting Go-friendly proto files
Remove google.protobuf references (for now) to avoid protoc-gen-go looking for non-existent files
Use type-to-file mappings when outputting field and extension names if referencing constructs in a different .proto file

TODO: Circular import references

### CODE REVIEW

**Comment by djkaty** [comment]
The second commit resolves all the remaining issues and should give a 'roughly' correct output with possibly one or two small things missing, which we can resolve later.

The output created now compiles correctly using protoc-gen-go with the current version of stove.

File: decompiler.cs [hunk 1]
```diff
@@ -489,7 +494,7 @@ class SilentOrbitTypeProcessor : TypeProcessor {
 					fieldType = FieldType.Bytes;
 					break;
 				default:
-					Console.WriteLine("unresolved type");
+                    Console.WriteLine("unresolved type for field '" + name + "' in " + result.Name.Text);
```
Comment (jleclanche) [inline decompiler.cs]: Bad indent change - also this alone can be a standalone commit.

File: protobuf.cs [hunk 2]
```diff
@@ -238,7 +238,7 @@ public class MessageNode : ILanguageNode, IImportUser {
 public class EnumNode : ILanguageNode {
 	public string Text {
 		get {
-			var result = String.Format("enum {0} {{\n", Name.Final);
+			var result = String.Format("enum {0} {{\n\toption allow_alias = true;\n", Name.Final);
```
Comment (jleclanche) [inline protobuf.cs]: Is there a way to only add that if we know it's required?

(And again, that should be a standalone commit)

Comment (djkaty) [inline protobuf.cs]: Yes but I just wanted to get it working, that's why I wrote 'quick fix' in the commit notes.

File: decompiler.cs (hunk 1 above)
Comment (djkaty) [inline decompiler.cs]: Visual Studio is retarded.

Comment (djkaty) [inline decompiler.cs]: Fixed in new commit

File: protobuf.cs (hunk 2 above)
Comment (djkaty) [inline protobuf.cs]: Fixed in new commit

**Comment by jleclanche** [comment]
1. Please update your brace style to [1TBS](https://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS)
2. Pull the blacklist update into its own commit - it's gtm
3. Pull the null reference fix in protobuf.cs into its own commit - gtm
4. Pull the `types` update into its own commit - NOT gtm, we'll separate it

**Comment by djkaty** [comment]
Did I miss anything?

I can do the types command line argument push when this one is through.

**Comment by jleclanche** [comment]
I won't land that types change because it breaks old builds - it can land after the cli arg.

**Comment by djkaty** [comment]
Do you want me to append that to this PR and amend the types commit, or open a new PR?

**Comment by djkaty** [comment]
I'm opening a new PR for the types change as I made it on its own branch
